File: content-solutions/binary-search/toy-workshop.java
class Solution {
    public long solution(int[] times, int m) {
        long fastest = Long.MAX_VALUE;
        for (int t : times) fastest = Math.min(fastest, t);
        long lo = 1, hi = fastest * m, answer = hi;
        while (lo <= hi) {
            long mid = lo + (hi - lo) / 2;
            long made = 0;
            for (int t : times) {
                made += mid / t;
                if (made >= m) break;
            }
            if (made >= m) {
                answer = mid;
                hi = mid - 1;
            } else lo = mid + 1;
        }
        return answer;
    }
}

File: content-solutions/binary-search/toy-workshop.py
def solution(times, m):
    lo, hi = 1, min(times) * m
    answer = hi
    while lo <= hi:
        mid = (lo + hi) // 2
        made = 0
        for t in times:
            made += mid // t
            if made >= m:
                break
        if made >= m:
            answer = mid
            hi = mid - 1
        else:
            lo = mid + 1
    return answer
